app/models/declaration_state.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class DeclarationState < ApplicationRecord
+  class << self
+    def submit!(participant_declaration:)
+      create!(participant_declaration: participant_declaration)
+    end
+
+    def void!(participant_declaration:)
+      create!(participant_declaration: participant_declaration, state: "voided")
+    end
+
+    def pay!(participant_declaration:)
+      create!(participant_declaration: participant_declaration, state: "paid")
+    end
+  end
```

Comment:
Why would we not have this on an instance of `ParticipantDeclaration`?

Reply:
At some point I want to move this to state transitions and this is a more logical place to handle those. Also the "reload" is a bit of a code smell to me, but that's probably more down to how the has_one and has_many linkages work.